server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/aws/AwsS3ObjectStorage.java

```diff
@@ -182,108 +161,67 @@ private File createTempFile(Blob blob) throws IOException {
-            return th instanceof UploadFailedException
-                && !doesBucketExist(bucketName);
-        }
-
-        private boolean failOnExistingBucket(Throwable th, ObjectStorageBucketName bucketName) {
-            return th instanceof UploadFailedException
-                && doesBucketExist(bucketName);
-        }
-
-        private boolean doesBucketExist(ObjectStorageBucketName bucketName) {
-            return getS3Client(configuration, getClientConfiguration())
-                .doesBucketExistV2(bucketName.asString());
-        }
-
-        private TransferManager getTransferManager() {
-            ClientConfiguration clientConfiguration = getClientConfiguration();
-            AmazonS3 amazonS3 = getS3Client(configuration, clientConfiguration);
-
-            return TransferManagerBuilder
-                    .standard()
-                    .withS3Client(amazonS3)
-                    .withMultipartUploadThreshold(MULTIPART_UPLOAD_THRESHOLD.getValue())
-                    .withExecutorFactory(() -> executorService)
-                    .withShutDownThreadPools(DO_NOT_SHUTDOWN_THREAD_POOL)
-                    .build();
-        }
-
-        private static AmazonS3 getS3Client(AwsS3AuthConfiguration configuration, ClientConfiguration clientConfiguration) {
-            return AmazonS3ClientBuilder
-                    .standard()
-                    .withClientConfiguration(clientConfiguration)
-                    .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(configuration.getAccessKeyId(), configuration.getSecretKey())))
-                    .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(configuration.getEndpoint(), null))
-                    .build();
-        }
+            getS3AsyncClient(configuration)
+                .map(client -> Mono.fromFuture(client
+                    .createBucket(CreateBucketRequest.builder()
+                        .bucket(bucketName.asString())
+                        .build())))
+                .block();
+        }
+
+        private boolean failOnNoSuchBucket(Throwable th) {
+            if (th instanceof NoSuchBucketException) {
+                NoSuchBucketException s3Exception = (NoSuchBucketException) th;
+                return NOT_FOUND_STATUS_CODE == s3Exception.statusCode()
+                    && BUCKET_NOT_FOUND_ERROR_CODE.equals(s3Exception.awsErrorDetails().errorCode());
+            }
 
-        private static ClientConfiguration getClientConfiguration() {
-            ClientConfiguration clientConfiguration = new ClientConfiguration();
-            clientConfiguration.setRetryPolicy(PredefinedRetryPolicies.getDefaultRetryPolicyWithCustomMaxRetries(MAX_ERROR_RETRY));
-            return clientConfiguration;
+            return false;
+        }
+
+        private static Mono<S3AsyncClient> getS3AsyncClient(AwsS3AuthConfiguration configuration) {
+            return Mono.fromCallable(() -> S3AsyncClient.builder()
```

Comment:
the client is supposed to be shared https://github.com/aws/aws-sdk-java-v2/blob/master/docs/BestPractices.md#reuse-sdk-client-if-possible